web/views/api/package_view.ex

```diff
@@ -21,9 +21,12 @@ defmodule HexWeb.API.PackageView do
     releases =
       Enum.map(releases, fn release ->
         version = to_string(release.version)
+        release = HexWeb.Repo.get_by(Release, version: version, package_id: package.id)
```

Comment:
this would result in N+1 queries which we definitely want to avoid for packages with a lot of releases. Could you adjust the code that grabs releases to also include retirement struct?